### ISSUE TITLE
Any progress on liberalizing the license?

### ISSUE THREAD

**Issue by dpacbach** [Wohlstand/libADLMIDI#226]
LGPL makes it hard to use!

**Comment by Wohlstand** [comment]
To have MIT/BSD/Zlib, I should replace all parts are was originally made by Bisqwit in his original ADLMIDI which are GPL. Right now, the only MIDI Sequencer and Chipset are MIT. OPL3 chip emulators in most are LGPL and GPL. The only Opal emulator is public domain, however, it doesn't support rhythm mode. Graduately I replacing old parta with new are more effective and I licensing them as MIT.

**Comment by Wohlstand** [comment]
It's not fast process.

**Comment by Wohlstand** [comment]
Speaking about original parts of Bisqwit are GPL licensed, I only can get the LGPL once I'll backport part of my work of 2014'th year into his original project. To get MIT, I should replace everything with my own, coded from scratch.

**Comment by dpacbach** [comment]
Ok... do you plan on doing that in the short-term future or should I give up hope on using libADLMIDI?

**Comment by Wohlstand** [comment]
Possible in near, however, I can't tell will that go fast or not, it's dependent on my free time I use for all of my projects.

**Comment by dpacbach** [comment]
@Wohlstand Do you see any possibility in the future to make this happen?

**Comment by Wohlstand** [comment]
Yes, I still think on that. The work still being not fast. Once upon I need to resume my work on embedded banks databse format change I almost completed on a second branch and continue the big work later. The new database format solves two goals:
- one another component with my own code I can license by any way
- more compact and better functionality of it, it will be able to store multibanks (non-GM banks like GS/XG and AIL banks with more than one melodic bank as an example at "The Lost Wikings" game)

**Comment by dpacbach** [comment]
Ok thanks; how many lines of code would you estimate have to be rewritten for the license goal to be met?

**Comment by Wohlstand** [comment]
Number of lines? Never cared about that...
So, I can tell better per component:
- MIDI Sequencer - done
- MIDI Synthesizer (a bridge between OPL3 chip and MIDI input) - didnt touched yet
- Banks database - in process, almost complete

**Comment by dpacbach** [comment]
Cool, let me know if I can do anything to help.
